### DISCUSSION TITLE
Is it possible to dynamically change the availible interfaces?

### DISCUSSION THREAD

**Discussion by GiantNuker** [hathach/tinyusb#707] [Q&A] [answered]
My use case is a config MSC to drop config modifications onto.
It would then parse them and get rid of the MSC.

Basically I just dont want the extra "drive" availible when I'm not using it

**Comment by hathach** [comment, marked as answer]
https://github.com/hathach/tinyusb/tree/master/examples/device/dynamic_configuration

**Reply by GiantNuker** [reply]
This requires unplugging and replugging the device right?
```c
// Invoked when received GET CONFIGURATION DESCRIPTOR
// Application return pointer to descriptor
// Descriptor contents must exist long enough for transfer to complete
uint8_t const * tud_descriptor_configuration_cb(uint8_t index)
{
  (void) index; // for multiple configurations
  return mode ? desc_configuraiton_1 : desc_configuration_0;
}
```
Can I force the host to reconnect somehow? (other than just timing out the host)

**Reply by hathach** [reply]
Try to use tud_disconnect then connect
